packages/contracts/contracts/optimistic-ethereum/ovm/SafetyChecker.sol

```diff
@@ -63,136 +35,102 @@ contract SafetyChecker is ContractResolver {
         view
         returns (bool)
     {
-        bool seenJUMP = false;
-        bool insideUnreachableCode = false;
-        uint256 ops = 0;
-        for (uint256 pc = 0; pc < _bytecode.length; pc++) {
+        // autogenerated by gen_safety_checker_constants.py
+        // number of bytes to skip for each opcode
+        uint256[8] memory skip = [
+          uint256(0x0001010101010101010101010000000001010101010101010101010101010000),
+          uint256(0x0100000000000000000000000000000000000000010101010101000000010100),
+          uint256(0x0000000000000000000000000000000001010101000000010101010100000000),
+          uint256(0x0203040500000000000000000000000000000000000000000000000000000000),
+          uint256(0x0101010101010101010101010101010101010101010101010101010101010101),
+          uint256(0x0101010101000000000000000000000000000000000000000000000000000000),
+          uint256(0x0000000000000000000000000000000000000000000000000000000000000000),
+          uint256(0x0000000000000000000000000000000000000000000000000000000000000000)];
+        // Mask to gate opcode specific cases
+        uint256 _opcodeProcMask = ~uint256(0xffffffffffffffffffffffe000000000fffffffff070ffff9c0ffffec000f001);
+        // Halting opcodes
+        uint256 _opcodeStopMask = ~uint256(0x6008000000000000000000000000000000000000004000000000000000000001);
+        // PUSH opcodes
+        uint256 _opcodePushMask = ~uint256(0xffffffff000000000000000000000000);
+
+        uint256 codeLength;
+        uint256 _pc;
+        assembly {
+            _pc := add(_bytecode, 0x20)
+        }
+        codeLength = _pc + _bytecode.length;
```

Comment:
Any reason why these local variables are not consistently named (`_varName` vs `name`)?

Reply:
good point! Fixed all the other variable names, but I just tried replacing `_pc` with `pc` and this does not work in assembly, where `pc` refers to the `PC` opcode